src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResponseResource.java

```diff
@@ -62,26 +59,66 @@ public ComplaintResponseResource(ComplaintResponseRepository complaintResponseRe
-        String entityName = complaintResponse.getComplaint().getComplaintType() == ComplaintType.MORE_FEEDBACK ? MORE_FEEDBACK_RESPONSE_ENITY_NAME : ENTITY_NAME;
+    public ResponseEntity<ComplaintResponse> lockComplaint(@PathVariable long complaintId) {
+        log.debug("REST request to create empty complaint response for complaint with id: {}", complaintId);
+        ComplaintResponse savedComplaintResponse = complaintResponseService.createEmptyComplaintResponse(complaintId);
+        // always remove the student from the complaint as we don't need it in the corresponding client use case
+        savedComplaintResponse.getComplaint().filterSensitiveInformation();
+        return new ResponseEntity<>(savedComplaintResponse, HttpStatus.CREATED);
+    }
 
+    /**
+     * POST /complaint-responses/complaint/:complaintId/refresh-lock: locks the complaint again by replace the empty complaint response
+     *
+     * @param complaintId - id of the complaint to lock again
+     * @return the ResponseEntity with status 201 (Created) and with body the empty complaint response
+     */
+    @PostMapping("/complaint-responses/complaint/{complaintId}/refresh-lock")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ComplaintResponse> refreshLockOnComplaint(@PathVariable long complaintId) {
+        log.debug("REST request to refresh empty complaint response for complaint with id: {}", complaintId);
+        ComplaintResponse savedComplaintResponse = complaintResponseService.refreshEmptyComplaintResponse(complaintId);
         // always remove the student from the complaint as we don't need it in the corresponding client use case
-        complaintResponse.getComplaint().filterSensitiveInformation();
+        savedComplaintResponse.getComplaint().filterSensitiveInformation();
+        return new ResponseEntity<>(savedComplaintResponse, HttpStatus.CREATED);
+    }
 
-        return ResponseEntity.created(new URI("/api/complaint-responses/" + savedComplaintResponse.getId()))
-                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, entityName, savedComplaintResponse.getId().toString())).body(savedComplaintResponse);
+    /**
+     * DELETE /complaint-responses/complaint/:complaintId/remove-lock: removes the lock on a complaint by removing the empty complaint response
+     *
+     * @param complaintId - id of the complaint to remove the lock for
+     * @return the ResponseEntity with status 200 (Ok)
+     */
+    @DeleteMapping("/complaint-responses/complaint/{complaintId}/remove-lock")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> removeLockFromComplaint(@PathVariable long complaintId) {
+        log.debug("REST request to remove the lock on the complaint with the id: {}", complaintId);
+        complaintResponseService.removeEmptyComplaintResponse(complaintId);
```

Comment:
seems that permission checks are missing here

Reply:
Done in the service. See:
private boolean isUserAuthorizedToRespondToComplaint(StudentParticipation participation, Complaint